pkg/gitops/flux/wait.go

```diff
@@ -50,30 +67,48 @@ func waitForFluxToStart(ctx context.Context, namespace string, timeout time.Dura
 		}
 		return nil
 	}
-	err := waitForPodToStart(namespace, "flux", 3030, "Flux", restConfig, cs, try)
+	err := portForwarding(namespace, "flux", 3030, "Flux", restConfig, cs, try)
 	return deployKey, err
 }
 
-func waitForHelmOpToStart(ctx context.Context, namespace string, timeout time.Duration, restConfig *rest.Config,
-	cs kubeclient.Interface) error {
-	try := func(rootURL string) error {
-		helmOpURL := rootURL + "healthz"
-		req, err := http.NewRequest("GET", helmOpURL, nil)
-		if err != nil {
-			return fmt.Errorf("failed to create request: %s", err)
-		}
-		healthzCtx, healtzhCtxCancel := context.WithTimeout(ctx, timeout)
-		defer healtzhCtxCancel()
-		req = req.WithContext(healthzCtx)
-		_, err = http.DefaultClient.Do(req)
+type tryFunc func(rootURL string) error
+
+func waitForDeploymentToStart(k8sClientSet kubeclient.Interface, namespace string, name string, timeout time.Duration) error {
+	watcher, err := k8sClientSet.AppsV1().Deployments(namespace).Watch(metav1.ListOptions{
+		FieldSelector: "metadata.name=" + name,
+	})
+	if err != nil {
 		return err
 	}
-	return waitForPodToStart(namespace, "helm-operator", 3030, "Helm Operator", restConfig, cs, try)
-}
 
-type tryFunc func(rootURL string) error
+	defer watcher.Stop()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return errors.Wrapf(err, "failed waiting for pod %q", name)
+			}
+			switch event.Type {
+			case watch.Added, watch.Modified:
+				deployment, ok := event.Object.(*v1.Deployment)
+				if !ok {
+					break
```

Comment:
If this condition is actually reached, it would be suggestive of a bug in the library. So I think we should return an error:
```suggestion
					return &typeAssertionError{&v1.Deployment{}, event.Object}
```

Reply:
Good point